src/Pact/Types/Continuation.hs

```diff
@@ -80,19 +81,26 @@ data Yield = Yield
     -- ^ Yield data from the pact continuation
   , _yProvenance :: !(Maybe Provenance)
     -- ^ Provenance data
+  , _ySourceChain :: !(Maybe ChainId)
   } deriving (Eq, Show, Generic)
 
 instance Arbitrary Yield where
   arbitrary = Yield <$> (genPactValueObjectMap RecurseTwice) <*> frequency
     [ (1, pure Nothing)
-    , (4, Just <$> arbitrary) ]
+    , (4, Just <$> arbitrary) ] <*> pure Nothing
 
 instance NFData Yield
-instance ToJSON Yield where toJSON = lensyToJSON 2
-instance FromJSON Yield where parseJSON = lensyParseJSON 2
+instance ToJSON Yield where
+  toJSON Yield{..} = object $
+      [ "data" .= _yData
+      , "provenance" .= _yProvenance ] ++
+      maybe [] (\c -> [ "source" .= c ]) _ySourceChain
+instance FromJSON Yield where
+  parseJSON = withObject "Yield" $ \o ->
+    Yield <$> o .: "data" <*> o .: "provenance" <*> o .:? "source"
 
 instance SizeOf Yield where
-  sizeOf (Yield dataYield prov) =
+  sizeOf (Yield dataYield prov _) =
```

Comment:
Why would you leave the size of the chain id out of this calculation?

Reply:
No particular reason, except that perhaps gas is supposed to represent some kind of increased utilization, whereas from a data standpoint, `Nothing` is equivalent to `Just ChainId` -- is `Nothing` worth 0? I didn't want to think about that question.